### ISSUE TITLE
Adguard for Mac continuously downloads data even if the computer in idle mode

### ISSUE THREAD

**Issue by evgeniy-ADG** [AdguardTeam/AdguardForMac#163]
**Ticket: 461686**

While no browsers active, Adguard starting to download excessive data which I don’t know what and why. I am having to sign off to stop this. After signing in I notice it does not continue to download.

<details>

<img width="309" alt="ekran resmi 2016-12-07 17 05 43" src="https://cloud.githubusercontent.com/assets/15856304/21762654/9006d842-d662-11e6-85ad-b302f2d5b2b6.png">

</details>

**Ticket: 1481462**

Hi, 2 days back, the office IT authority flagged my MAC for hogging the bandwidth up. As soon as I got to know of it, I opened the Activity Monitor up and realized that AdGuard was the culprit, which downloaded up to 10-15 GB worth content. I don’t have a diagnostic report with me right now but would like to enquire whether this has been a reported case before. Using an 11-inch MacBook Air (Mid 2013) running MacOS Sierra 10.12. Cheers!

<details>

Application version: 1.4.0
OS: 1012

Ads Blocked: 45995
Trackers Blocked: 114820
Fishing Blocked: 19

Antibanner filter: Enabled
Antiphishing filter: Enabled
Parental control: Disabled
Privacy protection: Enabled
HTTPS: Enabled
Assistant: Enabled
Web of trust: Enabled
Popup blocker: Enabled

Filters subscriptions:
ID=0 Name="User filter" Version=1.0.0.0
ID=2 Name="English filter" Version=1.1.4.70
ID=3 Name="Spyware filter" Version=1.0.9.99
ID=4 Name="Social media filter" Version=1.0.16.10
ID=14 Name="Annoyances filter" Version=1.0.8.46
ID=10 Name="Filter unblocking useful ads" Version=1.0.0.89
ID=6 Name="German filter" Version=1.0.29.91
ID=5 Name="Experimental filter" Version=1.0.26.41
ID=12 Name="Safari filter" Version=1.0.2.2
ID=101 Name="EasyList" Version=1.0.66.21
ID=118 Name="EasyPrivacy" Version=1.0.72.6
ID=123 Name="Fanboy's Social Blocking List" Version=1.0.66.23
ID=113 Name="Liste FR" Version=1.0.72.24
ID=120 Name="Norsk adblockliste" Version=1.0.0.1
ID=207 Name="Adblock Warning Removal List" Version=1.0.72.20
ID=208 Name="Malware Domains" Version=1.0.72.8

</details>

**Ticket: 1389893**

I've noticed that Adguard (1.3.0) sucks up a lot of bandwidth after being active for a while. Using Little Snitch 3.7 I can see that Adguard uses up to 2mbit/s of data for no apparent reason and without stopping until I disable Adguard and then re-enable it. Until I've cycled Adguard all my downloads and streams will be super slow (as I only have 10 Mbit's at home).

<details>

Application version: 1.3.0
OS: 1012

Antibanner filter: Enabled
Antiphishing filter: Enabled
Parental control: Disabled
Privacy protection: Enabled
HTTPS: Enabled
Assistant: Enabled
Web of trust: Disabled
Popup blocker: Enabled

Filters subscriptions:
ID=0 Name="User filter" Version=1.0.0.0
ID=2 Name="English filter" Version=1.1.2.70
ID=3 Name="Spyware filter" Version=1.0.9.44
ID=4 Name="Social media filter" Version=1.0.15.30
ID=14 Name="Annoyances filter" Version=1.0.7.59
ID=10 Name="Filter unblocking useful ads" Version=1.0.0.89
ID=1 Name="Russian filter" Version=1.0.44.6
ID=6 Name="German filter" Version=1.0.29.18
ID=8 Name="Dutch filter" Version=1.0.9.12
ID=9 Name="Spanish/Portuguese filter" Version=1.0.86.47
ID=12 Name="Safari filter" Version=1.0.1.81
ID=101 Name="EasyList" Version=1.0.65.37
ID=106 Name="EasyList Dutch" Version=1.0.70.72
ID=109 Name="EasyList Italy" Version=1.0.70.51
ID=207 Name="Adblock Warning Removal List" Version=1.0.70.65
ID=211 Name="Anti-Adblock Killer | Reek" Version=1.0.1.68
ID=208 Name="Malware Domains" Version=1.0.70.53

</details>

1486915

**Comment by vozersky** [comment]
I will give it more tests, but well, the traffic comes from some filtered app, Adguard doesn't really download anything except filter updates and it takes a really small amount of data. 

For example, I've played a video on YouTube, and opened LittleSnitch's NetworkMonitor afterwards. You can see that YouTube traffic was counted as Adguard's:
![image](https://cloud.githubusercontent.com/assets/17472907/21860319/374a0c48-d841-11e6-989b-f8f4194883ea.png) 

Can users do the same and check what server(s) was (were) involved in those cases of high traffic\bandwidth consumption?

**Comment by june8-o** [comment]
I unchecked all applications except Safari to test. Even while the Safari not open, I see that Adguard downloads data with full bandwidth. Weird thing is; if I turn Adguard off then on again, it does not continue to download.

**Comment by Rediwed** [comment]
I have been experiencing this issue for a few **months** now. No problem though, I will soon get unlimited 4G internet 💯

**Comment by vozersky** [comment]
Still no luck with reproducing such issue. 
Would appreciate a list of involved domains etc. from any network manager.

**Comment by Rediwed** [comment]
I must say, I haven't noticed this issue in the past week or so. Maybe I wasn't paying attention. I'll do my best!

**Comment by ameshkov** [comment]
@evgeniy-ADG first of all, please check that all these users are on v1.4.0.

**Comment by june8-o** [comment]
v1.4.0 (But the problem was the same with v.1.3)

**Comment by ameshkov** [comment]
Summoning @Stillness-2 here. @Stillness-2, do we need logs?

**Comment by anderworx** [comment]
No response from AdGuard? I submitted a ticket on this over a week ago. AdGuard 1.4.0 is consuming almost 20GB per day while I'm at work and the computer is idle, no apps running. I've had to turn it off to prevent going over my monthly Comcast/Xfinity data limit.

**Comment by Revod** [comment]
1.40 and have the same problem

**Comment by zebrum** [comment]
@Stillness-2 thoughts? do we need logs?

**Comment by june8-o** [comment]
Logs sent already. No one knows why and where Adguard downloads excessive data when it wants to. Even its developers.

**Comment by zebrum** [comment]
@june8-o 
we will appreciate if you collect some statistics for us.
Uncheck all apps in Preferences - Network - Automatically filter applications:
![2017-01-24 18 08 52](https://cloud.githubusercontent.com/assets/23637588/22252776/5e13fcfc-e260-11e6-8be5-eb8e849e8662.png)
use your mac as usual, but do not turn it off. and also run Activity monitor too.
after 3 days send us a screenshot with network information from activity monitor.

**Comment by june8-o** [comment]
@zebrum
AdGuard tech support already wanted me to do so. And I have already done this steps and sent screenshots  of network activity which shows that AdGuard downloads with max bandwidth continuously. 

Tech support also wanted me to disable auto macOS software updates, I also done so, but problem still continues.

So, what's next?

**Comment by zebrum** [comment]
@june8-o you disabled all applications except Safari as i recall

**Comment by june8-o** [comment]
Correct. Do you want me to disable also the Safari. If so, what should Adguard on my system?

**Comment by zebrum** [comment]
yep, we need you to disable all applications.

we can't reproduce this issue on our side, so we want to be sure it is Adguard itself downloading data. 
my mistake - you don't need run Activity monitor from the beginning, just to make screenshot. the main point is do not turn off Adguard.

**Comment by june8-o** [comment]
ok, I'll do so.

**Comment by june8-o** [comment]
I finished the test. While all applications disabled, after 72 hours (3 days) AdGuard downloaded only 8.7 MB data (that should be the filter updates). During this test, I haven't noticed any continuous downloads with my max bandwidth.

**Comment by Rediwed** [comment]
Alright. So I think I've found the issue.

I was just browsing Facebook when I noticed Adguard continue downloading a lot of data after I quit browsing. I tried to figure out was would be causing the problem, and it seems (to me) that adguard contacts Facebook's content distribution Network (fbcdn.com) and never closes that connection. In this case, a video is streamed from fbcdn.com. I've been able to reproduce the continued downloading behaviour consistently by opening Facebook, then watching a video (I also **clicked** on it, but that might not be necessary).

I even made a little video which reproduces the issue, check it here; https://www.youtube.com/watch?v=zF8VtzeLqGQ&feature=youtu.be (sorry for the stickies, it's for privacy).

Can anyone confirm behaviour? :D 💃

**Comment by Stillness-2** [comment]
@Rediwed, I confirm this. Thank you very much for research! :)

**Comment by june8-o** [comment]
@Rediwed I tested your steps. Watched a video on Facebook (video should be watched in Facebook, not linked to another site), quitted Safari. Adguard continued download data. Turned Adguard off, turned on, download stopped.

**Comment by Rediwed** [comment]
Great!!
I'm glad I was able to help!

This kind of personal support and quick fixes (i hope!! :D) makes me glad I pay for the product instead of it being a opensource thing.

**Comment by Stillness-2** [comment]
@june8-o @Rediwed, please check this build:
https://www.dropbox.com/s/74hmh731lfotn7u/Adguard-1.4.1.beta.dmg?dl=1

**Comment by june8-o** [comment]
@Stillness-2 Thank you. I fallowed with same steps previously taken with 1.4.0. Adguard did not download anything after quitting Safari.

**Comment by vozersky** [comment]
Oh, nice! thanks a lot @Rediwed!
I was trying to find the responsible domain among a bunch of video streaming services actually, but didn't check on FB vids, tbh.
Now we all know that it can be a good thing to browse FB at work :)

**Comment by Rediwed** [comment]
@Stillness-2 the beta seems to work perfectly!

**Comment by Rediwed** [comment]
@Stillness-2 I noticed that this beta version prevents you from succesfully loging into Facebook. I had to disable adguard temporary to be able to log in.

**Comment by vozersky** [comment]
@Rediwed @Stillness-2 can't reproduce the FB thing, can you send us some logs please, to check your filters at least .

**Comment by june8-o** [comment]
AdGuard warns me about 1.4.0 beta. Which one is newest?

**Comment by vozersky** [comment]
@june8-o the one you already have (with a fix)

**Comment by june8-o** [comment]
I suppose, it shouldn't warn for 1.4.0 beta over 1.4.1 beta.
